x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/get_allowed_fields_for_terms_query.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IndicesGetFieldMappingResponse } from '@elastic/elasticsearch/lib/api/types';
+import type { AllowedFieldsForTermsQuery, GetAllowedFieldsForTermQuery } from './types';
+
+const allowedFieldTypes = ['keyword', 'constant_keyword', 'wildcard', 'ip'];
+
+// Return map of fields allowed for term query
+export const getAllowedFieldForTermQueryFromMapping = (
+  indexMapping: IndicesGetFieldMappingResponse
+): { [key: string]: boolean } => {
+  const result: { [key: string]: boolean } = {};
+  const notAllowedTypes: string[] = [];
+
+  const indicies = Object.values(indexMapping);
+  indicies.forEach((index) => {
+    Object.entries(index.mappings).forEach(([field, fieldValue]) => {
+      Object.values(fieldValue.mapping).forEach((mapping) => {
+        const fieldType = mapping?.type;
+        if (!fieldType) return;
+
+        if (allowedFieldTypes.includes(fieldType) && !notAllowedTypes.includes(fieldType)) {
+          result[field] = true;
+        } else {
+          notAllowedTypes.push(fieldType);
+          // if we the field allowed in one index, but not allowed in another, we should delete it from result
+          delete result[field];
+        }
+      });
+    });
+  });
+
+  return result;
+};
+
+// Return map of fields allowed for term query for source and threat indices
```

Comment:
@nkhristinin,  I've seen you've made this change in some of places, but not others. Would be great if it can be done for other methods for consistency.